tests/TaskLineRenderer.test.Visualise_HTML_Full_task_-_full_mode.approved.html

```diff
@@ -0,0 +1,37 @@
+<li class="task-list-item plugin-tasks-list-item"
+    data-task-priority="medium"
+    data-task-created="past-far"
+    data-task-start="past-far"
+    data-task-scheduled="past-far"
+    data-task-due="past-far"
+    data-task-done="past-far"
+    data-task=""
+    data-line="0"
+    data-task-status-name="Todo"
+    data-task-status-type="TODO"><input class="task-list-item-checkbox" type="checkbox"
+    data-line="0">
+        <span class="tasks-list-text">
+        <span class="task-description">
+        <span></span></span>
```

Comment:
Sorry, I reviewed this in a hurry to get it merged and build on it...
This empty description was nagging at me overnight...

So I made this change to the test:

```diff
diff --git a/tests/TaskLineRenderer.test.ts b/tests/TaskLineRenderer.test.ts
index ec7f5a937..53b2ce1fc 100644
--- a/tests/TaskLineRenderer.test.ts
+++ b/tests/TaskLineRenderer.test.ts
@@ -602,11 +602,14 @@ describe('task line rendering', () => {
 describe('Visualise HTML', () => {
     async function renderAndVerifyHTML(task: Task, layoutOptions: LayoutOptions) {
         const parentRender = await createMockParentAndRender(task, layoutOptions);
+        const taskAsMarkdown = `<!--
+${task.toFileLineString()}
+-->\n\n`;
         const taskAsHTML = parentRender.innerHTML
             .replace(/ data-/g, '\n    data-')
             .replace(/<span/g, '\n        <span');
 
-        verifyWithFileExtension(taskAsHTML, 'html');
+        verifyWithFileExtension(taskAsMarkdown + taskAsHTML, 'html');
     }
 
     const fullTask = TaskBuilder.createFullyPopulatedTask();
```

Seeing the input in the approval test always makes it a lot easier to check the output...

And this unfortunately shows that that neither the description nor the tags are written out by our new test.

Reply:
It's also not displaying the urgency.

Reply:
> It's also not displaying the urgency.

Which we have decided is OK, as that is covered separately in `QueryRender.createTaskList()`...